### DISCUSSION TITLE
Submodule Version Update on Bitbucketcloud not working

### DISCUSSION THREAD

**Discussion by maebli** [renovatebot/renovate#23812] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest tag on docker hub

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

I never saw this working, however this pull request https://github.com/renovatebot/renovate/pull/17868 suggest it should work

### Wanted end result.

I have a repos A with renovate that updates 3 repos A,B and C

A is a submodule of B and C and I want to create automatic pull requests on B and C whenever there is a new version of A.

I read that private submodules are supported, so I am wondering what I'm doing wrong. I have enabled submodules in renovate.js

### What you tried so far.

added hostrules 

```
   "hostrules": [{
        "hostType": "bitbucket",
        "domainName": "bitbucket.org",
        "username": "$BITBUCKET_APP_USERNAME",
        "password": "$BITBUCKET_APP_PASSWORD"
    }
``` 

and also

```
    "hostRules": [
        {
            hostType: 'git-refs',
        }
``` 
and also tried

```
    "enabledManagers" : ["git-submodules"],
``` 

Tried with versions:

* latest 
* 36.42-full
* 35.159

### Relevant debug logs

<details><summary>Logs</summary>

```
WARN: Error mapping git submodules during extraction (repository=x, packageFile=.gitmodules)
       "err": {
         "task": {
           "commands": [
             "ls-remote",
             "--symref",
             "https://bitbucket.org/y",
             "HEAD"
           ],
           "format": "utf-8",
           "parser": "[function]"
         },
         "message": "fatal: could not read Username for 'https://bitbucket.org': No such device or address\n",
         "stack": "Error: fatal: could not read Username for 'https://bitbucket.org': No such device or address\n\n    at Object.action (/opt/containerbase/tools/renovate/36.41.0/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:42:29)\n    at PluginStore.exec (/opt/containerbase/tools/renovate/36.41.0/node_modules/simple-git/src/lib/plugins/plugin-store.ts:29:29)\n    at /opt/containerbase/tools/renovate/36.41.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:127:42\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/opt/containerbase/tools/renovate/36.41.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:124:14)\n    at GitExecutorChain.<anonymous> (/opt/containerbase/tools/renovate/36.41.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:100:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/opt/containerbase/tools/renovate/36.41.0/node_modules/simple-git/dist/cjs/index.js:55:24)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)"
       }
```

</details>

**Comment by maebli** [comment]
Possible duplicates

- https://github.com/renovatebot/renovate/discussions/23299
- https://github.com/renovatebot/renovate/discussions/21298
- https://github.com/renovatebot/renovate/discussions/21399
- https://github.com/renovatebot/renovate/discussions/21697

**Comment by DBS-ST-VIT** [comment]
Heya,
we also have the same issue. Nothing we tried (more or less all common solution attempts from StackOverflow and prior Issues) so far has worked 😕
